Provide information on externalized services (#5469)
circleci/circleci-docs: jekyll/_cci2/server-3-install-migration.adoc

* Stop your v2.19.x application
* Create a tarball of your v2.19.x application's PostgreSQL and Mongo databases
* Archive existing application data for Vault and CircleCI encryption/signing keys
* Export the tarball to your v3.x installation (exported data stores are stored in `circleci_export`, which may be useful for debugging purposes).
* Scale v3.x application deployments down to zero, and then to one
* Import the data from the previously exported tarball to your new v3.x instance

> NOTE: If your services are externalized then you may run `./migrate.sh --skip-databases` to skip Postgres and Mongo. In the event Postgres, Mongo, and Vault are all externalized, you simply need to scp your keys from `/data/circle/circleci-encryption-keys` on the services box.

Reviewer: ```suggestion
NOTE: If your services are externalized then you may run `./migrate.sh --skip-databases` to skip Postgres and Mongo. In the event Postgres, Mongo, and Vault are all externalized, you simply need to `scp` (secured copy) your keys from `/data/circle/circleci-encryption-keys` on the v2.19.x services machine.
```

Reviewer: Just some suggestions for clarity

> NOTE: If your services are externalized then you may run `./migrate.sh --skip-databases` to skip Postgres and Mongo. In the event Postgres, Mongo, and Vault are all externalized, you simply need to `scp` (secured copy) your keys from `/data/circle/circleci-encryption-keys` on the v2.19.x services machine.

Reviewer 2: Where do I copy them too?

```suggestion
NOTE: If your services are externalized then you may run `./migrate.sh --skip-databases` to skip Postgres and Mongo. In the event Postgres, Mongo, and Vault are all externalized, you need to `scp` (secured copy) your keys from `/data/circle/circleci-encryption-keys` on the v2.19.x services machine.
```

Reviewer 2: Where do I copy the keys to?

Author: Welp, I definitely fixed this but now I don't know where that commit went. Sorry. I will have to fix this tomorrow.

Author: Ah. I never `git push`d it. So embarrassing xD

> NOTE: If your services are externalized then you may run `./migrate.sh --skip-databases` to skip Postgres and Mongo. In the event Postgres, Mongo, and Vault are all externalized, you need to `scp` (secured copy) your keys from `/data/circle/circleci-encryption-keys` on the v2.19.x services machine. You may `cat` these files and upload their contents to the 3.x configuration page.

Reviewer: ```suggestion
NOTE: If your services are externalized then you can run `./migrate.sh --skip-databases` to skip Postgres and Mongo. In the event Postgres, Mongo, and Vault are all externalized, you need to `scp` (secured copy) your keys from `/data/circle/circleci-encryption-keys` on the v2.19.x services machine and then `cat` these files and upload their contents to the 3.x configuration page.
```

Reviewer: just style changes really

* Stop your v2.19.x application
* Create a tarball of your v2.19 application's PostgreSQL and Mongo databases
* Archive existing application data for Vault and CircleCI encryption/signing keys
* Export the tarball to your v3.x installation (exported data stores are stored in `circleci_export`, which may be useful for debugging purposes).
* Scale v3.x application deployments down to zero, and then to one
* Import the data from the previously exported tarball to your new v3.x instance

> NOTE: If you have externalized services then you can run `bash migrate.sh -v -p -m`. These `-v -p -m` flags will skip the migration of Vault, Postgres, and Mongo, respectively. Skipping all three will simply copy your keys from `/data/circle/circleci-encryption-keys` on the v2.19.x services machine, allowing you to `cat` these files and upload their contents to the 3.x configuration page.

Reviewer 2: ```suggestion
NOTE: If you have externalized services then you can run `bash migrate.sh -v -p -m`. These `-v -p -m` flags will skip the migration of Vault, Postgres, and Mongo, respectively. Skipping all three will copy your keys from `/data/circle/circleci-encryption-keys` on the v2.19.x services machine, allowing you to `cat` these files and upload their contents to the 3.x configuration page.
```
